src/domain/flush/flush.repository.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Inject,
+  Injectable,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { CacheRouter } from '../../datasources/cache/cache.router';
+import {
+  CacheService,
+  ICacheService,
+} from '../../datasources/cache/cache.service.interface';
+import { IConfigApi } from '../interfaces/config-api.interface';
+import { InvalidationPatternDto } from './entities/invalidation-pattern.dto.entity';
+import { InvalidationTarget } from './entities/invalidation-target.entity';
+import { IFlushRepository } from './flush.repository.interface';
+
+@Injectable()
+export class FlushRepository implements IFlushRepository {
+  constructor(
+    @Inject(CacheService) private readonly cacheService: ICacheService,
+    @Inject(IConfigApi) private readonly configApi: IConfigApi,
+  ) {}
+
+  async execute(pattern: InvalidationPatternDto): Promise<void> {
+    switch (pattern.invalidate) {
+      case InvalidationTarget.Chains:
+        return this.invalidateChains();
+      case InvalidationTarget.Contracts:
+        return this.invalidateContracts();
+      case InvalidationTarget.Tokens:
+        return this.invalidateTokens(pattern);
+    }
+  }
+
+  private async invalidateChains(): Promise<void> {
+    const chains = await this.configApi.getChains();
+    await this.cacheService.delete(CacheRouter.getChainsCacheDir());
```

Comment:
@fmrsabino what do you think about renaming this `ICacheService.delete` function to `ICacheService.deleteByKey`? I mean, the inner implementation of the function ignores the `field` part of the `CacheDir` passed in and it just deletes all the contents pointed by the `key` part, so I think it would be clearer for the consumers of ICacheService API 🤔

Reply:
Hmm, I'm not against keeping the current signature, but imagine a client of the API calling `delete(new CacheDir('foo', 'bar'))`. As a client, I think it's reasonable to expect that the field `bar` in within the key `foo` gets deleted, but not the whole key `foo`. Maybe we can just document that if a field is passed within the `CacheDir` it gets ignored and the function just deletes all the contents pointed by the key 🤔